publish.js

```diff
@@ -0,0 +1,293 @@
+#! /usr/bin/env node
+
+/*
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/fluid-project/first-discovery-server/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var publish = {};
+var path = require("path");
+var pkgPath = path.join(__dirname, "package.json");
+var pkg = require(pkgPath);
+var extend = require("extend");
+
+// The package.json file of the top level package which is
+// running this module.
+var modulePkgPath = path.join(process.cwd(), "package.json");
```

Comment:
These two global variables should also be hoisted into the publish.x object - right now we have the issue that the tests don't run from their own directory because they try to look for a package.json in their own directory. If this is a member then it can be overridden by the test fixture just like the other things that we mock.